pkg/spdx/implementation.go

```diff
@@ -232,19 +236,32 @@ func getImageReferences(referenceString string) ([]ImageReferenceInfo, error) {
 			return nil, fmt.Errorf("while calculating image digest: %w", err)
 		}
 
-		dig, err := name.NewDigest(
-			fmt.Sprintf(
-				"%s/%s@%s:%s",
-				tag.RegistryStr(), tag.RepositoryStr(),
-				imageDigest.Algorithm, imageDigest.Hex,
-			),
-		)
-		if err != nil {
-			return nil, fmt.Errorf("building single image digest: %w", err)
+		// If we could not turn the reference to digest then we synthesize one
+		if tok && !dok {
+			dig, err = name.NewDigest(
+				fmt.Sprintf(
+					"%s/%s@%s:%s",
+					tag.RegistryStr(), tag.RepositoryStr(),
+					imageDigest.Algorithm, imageDigest.Hex,
+				),
```

Comment:
I'd say we make this a utility function.

Reply:
Done!